src/app/mod.rs

```diff
@@ -1,12 +1,33 @@
-use std::time::Duration;
-
 use axum::http::StatusCode;
 use axum::response::{IntoResponse, Response};
 use axum::{extract::MatchedPath, http::Request, routing::get, Json, Router};
+use dotenvy::dotenv;
+use mongodb::{Client, Database};
 use serde_json::json;
+use std::sync::Arc;
+use std::time::Duration;
 use tower_http::classify::ServerErrorsFailureClass;
 use tower_http::trace::TraceLayer;
 
+#[derive(Clone)]
+pub struct State {
+    pub db: Arc<Database>,
+}
+
+impl State {
+    /// # Errors
+    ///
+    /// Will return 'Err' if there is no `DATABASE_URL` environment variable or the app
+    /// cannot connect to the database
+    pub async fn new() -> anyhow::Result<Self> {
+        dotenv().ok();
+        let database_url = std::env::var("DATABASE_URL")?;
```

Comment:
Can you move this logic to config.rs for now so that when I change this to key rotation I don't have to change the logic outside of the config file